infra.bs

```diff
@@ -174,6 +174,78 @@ caller's algorithm. Using prose the caller has the ability to "catch" the except
 another action.
 
 
+<h3 id=algorithm-conditional-abort>Conditional abort</h3>
+
+<p>Sometimes it is useful to stop performing a series of steps once some condition becomes true.
+
+<p>To do this, state that a given series of steps will <dfn export>abort when</dfn> a specific
+<var>condition</var> is reached. This indicates that the specified steps must be evaluated, not
+as-written, but by additionally inserting a step before each of them that evaluates
+<var>condition</var>, and if <var>condition</var> evaluates to true, skips the remaining steps.
+
+<p>In such algorithms, the subsequent step can be annotated to run <dfn export>if aborted</dfn>, in
+which case it must run if the <var>condition</var> of the preceding <a>abort when</a> step evaluated
+to true.
```

Comment:
Might be clearer to say "if the preceding steps were skipped due to *condition* becoming true".

The ambiguity is if the preceding steps completed, but *condition* became true before the "if aborted" line.